otherlibs/unix/errmsg.c

```diff
@@ -21,5 +21,6 @@
 CAMLprim value unix_error_message(value err)
 {
   int errnum = code_of_unix_error(err);
+  /* strerror is safe here as we have the domain runtime lock */
```

Comment:
Maybe I misunderstanding how this works, but what if there are multiple domains running and raising different Unix_error exceptions? I thought there is no more global runtime lock, and the domain runtime lock would only apply to threads within the same domain, but not cross-domain, is that right?
I'd suggest to call caml_strerror anyway to be safe.

Reply:
Indeed, this works most of the time but not in the case of a simultaneous strerror with an unknown error type. Fixed in 84e6f51b6309e5674c4dd1729ac1d60c1739a45d